oi/type_graph/TopoSorter.cpp

```diff
@@ -73,6 +73,10 @@ void TopoSorter::visit(Class& c) {
   }
 }
 
+void TopoSorter::visit(Primitive& p) {
```

Comment:
Why include primitives in the sorting now?

Reply:
They need a name provided for them, e.g. a `std::vector` of `int32_t` will use `NameProvider<int32_t>::names`. I would have iterated over the primitive kinds to do this, but there's the `case Kind::Float80: abort();` which means that's not possible. Having them appear in the output to get named seem best.